src/main/java/org/concordion/internal/listener/ResourcesFactory.java

```diff
@@ -0,0 +1,191 @@
+package org.concordion.internal.listener;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.concordion.api.Resources;
+import org.concordion.api.Resources.InsertType;
+import org.concordion.api.Fixture;
+import org.concordion.api.Resource;
+import org.concordion.api.extension.ConcordionExtender;
+import org.concordion.internal.util.IOUtil;
+
+/**
+ * Copy resources to the destination when specification is built.  Resources are copied to the same relative path as they are in the project.
+ * Currently only supports resources on the file path.  Support for resources in Jars may come at a later date but will require java 1.7.
+ * 
+ * @author sumnera
+ */
+public class ResourcesFactory {
+	private boolean includeDefaultStyling = true;
+	private List<ResourceToCopy> sourceFiles = new ArrayList<ResourceToCopy>();
+		
+	public boolean includeDefaultStyling() {
+		return this.includeDefaultStyling;
+	}
+	
+	public List<ResourceToCopy> getFiles() {
+		return this.sourceFiles;
+	}
+	
+	public ResourcesFactory(ConcordionExtender builder, Fixture fixture) {
+		
+		File root = getRootPath(fixture.getFixtureClass());
```

Comment:
We can't assume that there is one folder for the class path. In many cases, the class path will contain multiple folders. For instance, the tests are currently failing in Gradle, since Gradle uses separate folders on the classpath for each source directory set. This is shown by running the following test:

```
    @Test
    public void printResourceFolders() {
        int folderCount = 0;
        Enumeration<URL> resources;
        try {
            resources = this.getClass().getClassLoader().getResources("");
            while (resources.hasMoreElements()) {
                URL url = (URL) resources.nextElement();
                System.out.println(url);
                folderCount++;
            }
        } catch (IOException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        Assert.assertEquals(1, folderCount);
    }
```

When run using Gradle, I see the output:

```
file:/local/work/concordion/andrew-sumner/build/classes/test/
file:/local/work/concordion/andrew-sumner/build/resources/test/
file:/local/work/concordion/andrew-sumner/build/classes/main/
file:/local/work/concordion/andrew-sumner/build/resources/main/
file:/local/work/concordion/andrew-sumner/build/classes/testDummies/
file:/local/work/concordion/andrew-sumner/build/resources/testDummies/
```
The code will need to cater for multiple folders on the classpath, combining resource files from all folders into the list of resources to copy.

Eclipse combines the output into a single folder by default, which is why the tests are working in Eclipse.